src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.cs

```diff
@@ -0,0 +1,1244 @@
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                char leftValue = left[i];
+                char rightValue = right[i];
+
+                // The elements must be an exact match, or they must be
+                // equivalent when converted to lowercase.
+
+                if ((leftValue != rightValue) && (ToLower(leftValue) != ToLower(rightValue)))
+                {
+                    return false;
+                }
+            }
+
+            return true; // no mismatches seen
+        }
+
+        /*
+         * Compares an ASCII byte buffer and an ASCII char buffer for equality, optionally treating
+         * [A-Z] and [a-z] as equal. Returns false if the ASCII byte buffer contains any non-ASCII
+         * data or if the char buffer contains any element in the range [ 0080 .. FFFF ], as we
+         * wouldn't know what encoding to use to perform the transcode-then-compare operation.
+         */
+
+        /// <summary>
+        /// Returns a value stating whether the contents of two ASCII text buffers are equal.
+        /// One buffer is provided as 8-bit <see cref="byte"/>s. The other buffer is provided
+        /// as 16-bit <see cref="char"/>s.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+        {
+            if (left.Length != right.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                byte leftValue = left[i];
+                if (!IsAscii(leftValue) || leftValue != right[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a value stating whether the contents of two ASCII text buffers are equal.
+        /// One buffer is provided as 8-bit <see cref="byte"/>s. The other buffer is provided
+        /// as 16-bit <see cref="char"/>s.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="right"/> is <see langword="null"/>.</exception>
+        public static bool Equals(ReadOnlySpan<byte> left, string right)
+        {
+            if (right is null)
+            {
+                throw new ArgumentNullException(nameof(right));
```

Comment:
In other methods (in this class) a throw-helper is used. E.g. 
https://github.com/dotnet/runtime/blob/bff016dfaca9db61027beac0c270c0de6dd2c829/src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.cs#L369

Should this be consistent?